airflow/providers/sftp/operators/sftp.py

```diff
@@ -129,7 +143,7 @@ def __init__(
                 )
                 self.sftp_hook = SFTPHook(ssh_hook=self.ssh_hook)
 
-    def execute(self, context: Any) -> str | None:
+    def execute(self, context: Any) -> list[str] | None:
```

Comment:
I wonder if it’d be a good idea to still return `str` if a str is passed (instead of `list[str]`). This would avoid breaking compatibility.

Reply:
Ah, good point - I had considered that at one point and forgot to incorporate it.  Hmm, what do you think - set a flag if I converted a `str` to a `list[str]` or just make `execute` return a `str` if `len(list)` == 1?

Reply:
I'm leaning toward setting a flag since that would be the only way to keep it consistent.  I'll push shortly - adding a test to cover this scenario.

Reply:
OK @uranusjr - see pauldalewilliams/airflow@9157bbc
Look better?